api/cds.proto

```diff
@@ -161,24 +161,22 @@ message Cluster {
   //   verification.
   UpstreamTlsContext tls_context = 11;
 
-  oneof protocol_options {
-    // [#not-implemented-hide:]
-    TcpProtocolOptions tcp_protocol_options = 12;
-
-    // Additional options when handling HTTP1 requests.
-    Http1ProtocolOptions http_protocol_options = 13;
-
-    // Even if default HTTP2 protocol options are desired, this field must be
-    // set so that Envoy will assume that the upstream supports HTTP/2 when
-    // making new HTTP connection pool connections. Currently, Envoy only
-    // supports prior knowledge for upstream connections. Even if TLS is used
-    // with ALPN, `http2_protocol_options` must be specified. As an aside this allows HTTP/2
-    // connections to happen over plain text.
-    Http2ProtocolOptions http2_protocol_options = 14;
-
-    // [#not-implemented-hide:]
-    GrpcProtocolOptions grpc_protocol_options = 15;
-  }
+  // [#not-implemented-hide:]
+  TcpProtocolOptions tcp_protocol_options = 12;
+
+  // Additional options when handling HTTP1 requests.
+  Http1ProtocolOptions http_protocol_options = 13;
+
+  // Even if default HTTP2 protocol options are desired, this field must be
+  // set so that Envoy will assume that the upstream supports HTTP/2 when
+  // making new HTTP connection pool connections. Currently, Envoy only
+  // supports prior knowledge for upstream connections. Even if TLS is used
+  // with ALPN, `http2_protocol_options` must be specified. As an aside this allows HTTP/2
+  // connections to happen over plain text.
+  Http2ProtocolOptions http2_protocol_options = 14;
+
+  // [#not-implemented-hide:]
+  GrpcProtocolOptions grpc_protocol_options = 15;
```

Comment:
This is something we probably want medium-long term, so should keep. There are gRPC specific options, like how much to buffer for gRPC retries, that could belong here (I think; can't remember where we left that particular one @alyssawilk @mattklein123).

Reply:
Like my other comment, I would personally kill this until it's needed later. As it's no longer mutually exclusive, GrpcProtocolOptions having an embedded Http2ProtocolOptions I don't think makes sense?

Reply:
I tend to lean towards treating gRPC as H2 until we have a concrete reason not to (which we probably will, eventually) and add it back then

Reply:
+1

Reply:
+1 (but still suspicious we'll end up wanting it at some point).